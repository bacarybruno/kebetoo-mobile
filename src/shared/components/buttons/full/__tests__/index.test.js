import { render, fireEvent } from 'react-native-testing-library'

import setupTest from 'Kebetoo/src/config/jest-setup'

import FullButton from '../index'

const givenFullButton = setupTest(FullButton, render)({
  text: 'New',
  style: { backgroundColor: 'red', color: 'white' },
  onPress: jest.fn(),
})

it('renders FullButton', () => {
  const { wrapper } = givenFullButton()
  expect(wrapper.toJSON()).toMatchSnapshot()
})

it('handles button press', () => {
  const { wrapper, props } = givenFullButton()
  fireEvent.press(wrapper.getByText(props.text.toUpperCase()))
  expect(props.onPress).toBeCalledTimes(1)
})
