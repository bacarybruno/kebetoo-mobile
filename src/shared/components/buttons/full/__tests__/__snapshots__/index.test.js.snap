// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FullButton 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": 15,
      "color": "white",
      "elevation": 2,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "rgba(0, 0, 0, 0.8)",
      "shadowOffset": Object {
        "x": 0,
        "y": 1,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Helvetica",
        },
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "normal",
          "opacity": 1,
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  >
    New
  </Text>
</View>
`;
